nixos/modules/services/audio/alsa.nix

```diff
@@ -80,6 +88,9 @@ in
         };
       };
 
+    hardware.firmware = mkIf config.sound.enableAlsaFirmware
+      [ "${pkgs.alsa-firmware}/lib/firmware" ];
+
```

Comment:
first, thanks for doing this.
second, I'm a nix noobie but shouldn't you also add it to systemPackages here?